src/telegram2discord/endwares.ts

```diff
@@ -152,13 +153,55 @@ export const relayMessage = (ctx: TediCrossContext) =>
 			const channel = await fetchDiscordChannel(ctx.TediCross.dcBot, prepared.bridge);
 
 			let dcMessage = null;
+
+			// Check if the message is a reply and get the id of that message on Discord
+			let replyId = 0;
+			const messageReference = ctx.tediCross.message?.reply_to_message;
+
+
+			if (typeof messageReference !== "undefined") {
+				const referenceId = messageReference?.message_id;
+				if (typeof referenceId !== "undefined") {
+					console.log("==== telegram2discord/endware.ts reply ====");
+					console.log("referenceId: " + referenceId);
+					console.log("prepared.bridge.name: "+ prepared.bridge.name);
+					[replyId] = ctx.TediCross.messageMap.getCorrespondingReverse(MessageMap.DISCORD_TO_TELEGRAM, prepared.bridge, referenceId as string);
+					console.log("d2t replyId: " + replyId);
+					if (isNaN(replyId)) {
+						[replyId] = ctx.TediCross.messageMap.getCorresponding(MessageMap.TELEGRAM_TO_DISCORD, prepared.bridge, referenceId as string);
+						console.log("t2d replyId: " + replyId);
+					}
+				}
+			}
+
+			let messageToReply: any;
+
+			if (replyId !== 0) {
+				messageToReply = await channel.messages.fetch((replyId as unknown as string)).catch((err: Error) => {
+					console.error(`Could not find Message ${replyId} in Discord Channel ${channel.id} on bridge ${prepared.bridge.name}: ${err.message}`);
+					throw err;
+				}) as unknown as Promise<Message>;
+				console.log("messageToReply.id: " + messageToReply.id);
+			}
+
 			// Send the attachment first, if there is one
 			if (!R.isNil(prepared.file)) {
 				try {
 					dcMessage = await channel.send({
 						content: R.head(chunks),
 						files: [prepared.file]
 					});
+					// if (replyId === 0 || replyId === undefined || messageToReply === undefined) {
+					// 	dcMessage = await channel.send({
+					// 		content: R.head(chunks),
+					// 		files: [prepared.file]
+					// 	});
+					// } else {
+					// 	dcMessage = await messageToReply.reply({
+					// 		content: R.head(chunks),
+					// 		files: [prepared.file]
+					// 	});
+					// }
```

Comment:
```suggestion
```

Reply:
This was the T -> D code, uncommenting it after doing some extra checks, actually, I'll put it back in use since at least T to T replies work with discord replies, it's T to D replies that don't reflect on discord replies due to lack of getCorrespondingReverse() functioning as intended.